roles/edpm_network_config/tasks/main.yml

```diff
@@ -14,136 +14,25 @@
-      ((os_net_config_returncode_slurp.content | b64decode | int) != 0)) or
-       (not os_net_config_returncode_stat.stat.exists)
+- name: Load network system role
   block:
-    - name: Create /etc/os-net-config directory
-      become: true
-      ansible.builtin.file:
-        path: /etc/os-net-config
-        state: directory
-        recurse: true
-
-    - name: Create os-net-config mappings from lookup data
-      edpm_os_net_config_mappings:
-        net_config_data_lookup:
-          "{{ edpm_network_config_os_net_config_mappings }}"
-      when: not ansible_check_mode|bool
-      register: os_net_config_mappings_result
-
-    - name: Write os-net-config mappings file /etc/os-net-config/mapping.yaml
-      ansible.builtin.copy:
-        content: "{{ os_net_config_mappings_result.mapping | to_nice_yaml }}"
-        dest: /etc/os-net-config/mapping.yaml
-        mode: 0644
-      when: os_net_config_mappings_result.changed|bool
-
-    - name: Manage NetworkConfig with edpm_os_net_config module
-      block:
-        - name: Remove /var/lib/edpm-config/scripts directory
-          ansible.builtin.file:
-            path: /var/lib/edpm-config/scripts
-            state: absent
-
-        - name: Run NetworkConfig with edpm_os_net_config
-          ansible.builtin.include_tasks: os_net_config.yml
-
-    - name: Write rc of NetworkConfig script
-      ansible.builtin.copy:
-        content: "{{ network_config_result.rc }}"
-        dest: /var/lib/edpm-config/os-net-config.returncode
-        mode: 0644
-      when:
-        - network_config_result.rc is defined
-
-    # LP Bug: https://bugs.launchpad.net/edpm/+bug/1958332
-    - name: Disable cloud-init network config
-      ansible.builtin.copy:
-        content: |
-          network:
-            config: disabled
-        dest: /etc/cloud/cloud.cfg.d/99-edpm-disable-network-config.cfg
-        mode: 0644
-      when:
-        - network_config_result.rc is defined
-        - network_config_result.rc == 0
-        - cloud_init_exists.stat.exists
-        - cloud_init_exists.stat.isdir
-
-    # os-net-config currently relies on the legacy network
-    # so we need to ensure it's enabled on boot. This should
-    # be removed when we switch to NetworkManager or replaced
-    # with something that ensures NetworkManager is enabled.
-    - name: Ensure network service is enabled
-      ansible.builtin.systemd:
-        name: network
-        enabled: true
-      when:
-        - edpm_network_config_manage_service
-        - not edpm_network_config_nmstate|bool
-        - not ansible_check_mode|bool
+    - name: Retrieve role name
+      ansible.builtin.set_fact:
+        systemrolename:
+          "{{ ansible_facts['distribution'] | lower }}.
+          {%- if ansible_facts['distribution'] == 'RedHat' -%}
+            rhel
+          {%- else -%}
+            linux
+          {%- endif -%}
+          _system_role.network"
+      delegate_to: localhost
+    - name: Configure network with nmstate
```

Comment:
Name of the task should indicate that the role has been dynamically selected. It may cut down on the time needed to troubleshoot things.

Reply:
you're right, let me fix this